pkg/planner/core/memtable_infoschema_extractor.go

```diff
@@ -720,26 +733,27 @@ func listTablesForEachSchema(
 // returns a schema slice and a table slice that has the same length.
 // Note that input arg "tableSlice" will be changed in place.
 func findSchemasForTables(
-	ctx context.Context,
 	is infoschema.InfoSchema,
-	schemas []pmodel.CIStr,
+	extractedSchemas []pmodel.CIStr,
+	unspecified bool,
 	tableSlice []*model.TableInfo,
 ) ([]pmodel.CIStr, []*model.TableInfo, error) {
 	schemaSlice := make([]pmodel.CIStr, 0, len(tableSlice))
 	for i, tbl := range tableSlice {
+		dbInfoName, ok := is.SchemaNameByTableID(tbl.DBID)
+		if !ok {
+			continue
+		}
+		if len(extractedSchemas) == 0 && unspecified {
```

Comment:
Seems like if `unspecified == true`, then `len(extractedSchemas)` is always `nil`.